src/amo/components/AutoSearchInput/index.js

```diff
@@ -0,0 +1,358 @@
+      return;
+    }
+
+    this.dispatchAutocompleteStart({ filters });
+  }
+
+  getSuggestions(): Array<SuggestionType> {
+    if (this.props.loadingSuggestions) {
+      // 10 is the maximum number of results returned by the API
+      return Array(10).fill({
+        iconUrl: getAddonIconUrl(),
+        name: this.props.i18n.gettext('Loading'),
+        loading: true,
+        url: undefined,
+      });
+    }
+
+    return this.props.suggestions;
+  }
+
+  handleSearch = (event: SyntheticEvent<any>) => {
+    event.preventDefault();
+
+    if (this.searchInput) {
+      // When submitting the form to view all search results, blurring
+      // the input will hide the suggestion results menu.
+      //
+      // TODO: We may be able to blur this without relying on a ref soon:
+      // https://github.com/moroshko/react-autosuggest/issues/370
+      //
+      // The method for obtaining this ref is undocumented but the
+      // library author suggests it in this comment:
+      // https://github.com/moroshko/react-autosuggest/issues/158#issuecomment-219322960
+      this.searchInput.blur();
+    }
+
+    const filters = this.createFiltersFromQuery(this.state.searchValue);
+    this.props.onSearch(filters);
+  }
+
+  handleSearchChange = (
+    event: SyntheticEvent<HTMLInputElement>,
+    { newValue }: OnSearchChangeParams
+  ) => {
+    const searchValue = newValue || '';
+    if (searchValue.trim().length <= SEARCH_TERM_MAX_LENGTH) {
+      this.setState({ searchValue });
+    }
+  }
+
+  handleSuggestionSelected = (
+    e: SyntheticEvent<any>,
+    { suggestion }: {| suggestion: SuggestionType |}
+  ) => {
+    e.preventDefault();
+
+    if (this.props.loadingSuggestions) {
+      log.debug('Ignoring a click on the suggestion while loading');
+      return;
+    }
+
+    this.setState({ autocompleteIsOpen: false, searchValue: '' });
+    this.props.onSuggestionSelected(suggestion);
+  }
+
+  renderSuggestion = (suggestion: SuggestionType) => {
+    const { i18n, loadingSuggestions, selectSuggestionText } = this.props;
+    const { name, iconUrl } = suggestion;
+
+    const arrowAlt =
+      selectSuggestionText || i18n.gettext('Go to the add-on page');
+    return (
+      <SearchSuggestion
+        name={name}
+        iconUrl={iconUrl}
+        loading={loadingSuggestions}
+        arrowAlt={arrowAlt}
+      />
+    );
+  }
+
+  render() {
+    const {
+      errorHandler,
+      i18n,
+      inputName,
+      inputPlaceholder,
+    } = this.props;
+
+    const autocompleteIsOpen = this.state.autocompleteIsOpen &&
+      // This prevents the input to look like Autosuggest is open when
+      // there are no results coming from the API.
+      this.getSuggestions().length > 0;
+
+    const inputProps = {
+      className: 'AutoSearchInput-query',
+      maxLength: SEARCH_TERM_MAX_LENGTH,
+      name: inputName,
+      onChange: this.handleSearchChange,
+      placeholder: inputPlaceholder || i18n.gettext('Find add-ons'),
```

Comment:
`i18n` objects can never be used as static default props because those get executed when the class code is parsed. This is too early to access `i18n` and doing so would result in the wrong localization anyway. Here is an example: https://codepen.io/kumar303/pen/LdGVoV?editors=1111

Reply:
Oh snap, right. That makes sense, but I just totally forgot. Might warrant a comment reminding someone looking at the code, because I knew this but forgot.